tikv config: add in memory pessismitic lock size limit config (#18916)
pingcap/docs: tikv-configuration-file.md

+ Default value: `true`
+ Note that `in-memory` takes effect only when the value of `pipelined` is `true`.

### `in-memory-peer-size-limit` <span class="version-mark">New in v8.4.0</span>

> + Control the memory size limit of pessimistic locks for one region. The pessimistic locks would be written persistently if the memory usage exceeds the limit.

Reviewer: ```suggestion
+ Controls the memory usage limit for [in-memory pessimistic locks](/pessimistic-transaction.md#in-memory-pessimistic-lock) in a Region. When this limit is exceeded, TiKV writes pessimistic locks persistently.
```

+ Default value: `512KiB`
+ Unit: KiB|MiB|GiB

### `in-memory-instance-size-limit` <span class="version-mark">New in v8.4.0</span>

> + Control the memory size limit of pessimistic locks for one TiKV instance. The pessimistic locks would be written persistently if the memory usage exceeds the limit.

Reviewer: ```suggestion
+ Controls the memory usage limit for [in-memory pessimistic locks](/pessimistic-transaction.md#in-memory-pessimistic-lock) in a TiKV instance. When this limit is exceeded, TiKV writes pessimistic locks persistently.
```
